pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -385,6 +391,18 @@ type containerToKillInfo struct {
 	message string
 }
 
+// containerToUpdateInfo contains necessary information to update a container's resources.
+type containerToUpdateInfo struct {
+	// Value of the index of the specific container in pod.Spec.Containers.
+	specIndex int
+	// Status of the container in pod.Spec.ContainerStatuses.
+	apiContainerStatus *v1.ContainerStatus
+	// Status of the runtime container
+	kubeContainerStatus *kubecontainer.ContainerStatus
+	// Resize requires restart
+	restart *bool
```

Comment:
It looks like restart==true means "skip the update".  Can we just not add the container to the list of containers to update instead of adding this field?

Reply:
Let me see if I can refactor this -- in effect undo append/prepend into ContainersToUpdate array if memory resize can be done without restart, but CPU comes up as restart.

Reply:
i was able to fix this quite easily, it will be fixed in the next commit.